src/System.IO.Pipelines/tests/FlushAsyncCancellationTests.cs

```diff
@@ -307,6 +307,35 @@ public void GetResultThrowsIfFlushAsyncTokenFiredAfterCancelPending()
             Assert.False(awaiterIsCompleted);
             Assert.True(onCompletedCalled);
         }
+
+        [Fact]
+        public async Task FlushAsyncThrowsIfPassedCanceledCancellationTokenAndPipeIsAbleToComplete()
+        {
+            var cancellationTokenSource = new CancellationTokenSource();
+            cancellationTokenSource.Cancel();
+
+            // AsTask is important here, it validates that we are calling completion callback
+            // and not only setting IsCompleted flag
+            var task = Pipe.Reader.ReadAsync().AsTask();
+            bool threwException = false;
+
+            try
+            {
+                await Pipe.Writer.FlushAsync(cancellationTokenSource.Token);
+            }
+            catch (OperationCanceledException)
+            {
+                threwException = true;
+            }
+            finally
+            {
+                Pipe.Writer.Complete();
+            }
+
+            Assert.True(threwException);
+            Assert.True(task.IsCompleted);
```

Comment:
I think IsCompleted is true for faulted/cancelled tasks, so this is guaranteed to be true.

Reply:
Nvm. I didn't read this test correctly. I now see this is the read task.